### ISSUE TITLE
Error with Pinpoint Workshop

### ISSUE THREAD

**Issue by pscheri** [aws-samples/retail-demo-store#35]
Hi, I'm following the Workshop notebooks and while executing the "Create Pinpoint Recommender Configuration" step inside 4.1-Pinpoint notebook I receive the following error:

`---------------------------------------------------------------------------
BadRequestException                       Traceback (most recent call last)
<ipython-input-35-52824d936032> in <module>()
     15         'RecommendationProviderUri': personalize_campaign_arn,
     16         'RecommendationTransformerUri': lambda_function_arn,
---> 17         'RecommendationsPerMessage': 4
     18     }
     19 )

~/anaconda3/envs/python3/lib/python3.6/site-packages/botocore/client.py in _api_call(self, *args, **kwargs)
    314                     "%s() only accepts keyword arguments." % py_operation_name)
    315             # The "self" in this scope is referring to the BaseClient.
--> 316             return self._make_api_call(operation_name, kwargs)
    317 
    318         _api_call.__name__ = str(py_operation_name)

~/anaconda3/envs/python3/lib/python3.6/site-packages/botocore/client.py in _make_api_call(self, operation_name, api_params)
    624             error_code = parsed_response.get("Error", {}).get("Code")
    625             error_class = self.exceptions.from_code(error_code)
--> 626             raise error_class(parsed_response, operation_name)
    627         else:
    628             return parsed_response

BadRequestException: An error occurred (BadRequestException) when calling the CreateRecommenderConfiguration operation: The specified ARN 'arn:aws:lambda:us-east-1:++++++++:function:RetailDemoStorePinpointRecommender' of the lambda function isn’t valid. Check the ARN of the function and permissions in its policy.
`
Thanks in advance!

**Comment by james-jory** [comment]
Thank you for the bug report, @pscheri. We are investigating internally and will provide an update as soon as possible.

**Comment by james-jory** [comment]
This issue was caused by a recent change in the Pinpoint service in how Lambda resource policies must be setup for recommender configurations. Until we can get a permanent fix applied, you can resolve the issue for your local deployment by running the following command (replacing `[REGION]` and `[ACCOUNT_ID]` with the AWS region and account ID for your deployment).

```
$ aws lambda add-permission \
--function-name arn:aws:lambda:[REGION]:[ACCOUNT_ID]:function:RetailDemoStorePinpointRecommender \
--statement-id PinpointPermission2 \
--action lambda:InvokeFunction \
--principal pinpoint.amazonaws.com \
--source-arn "arn:aws:mobiletargeting:[REGION]:[ACCOUNT_ID]:*" \
--region [REGION]
```

Then try running the notebook cell again where the error occurred before.

**Comment by james-jory** [comment]
To apply the above fix (#37) to an existing Retail Demo Store deployment, you can update your CloudFormation stack. Once updated, re-execute the notebook cell.
